schema.go

```diff
@@ -93,7 +93,9 @@ type IndexSchema struct {
 	// exist from a structure.
 	AllowMissing bool
 
-	Unique  bool
+	Unique            bool
+	EnforceUniqueness bool
```

Comment:
I think the naming of this is a bit confusing, like, why wouldn't the Unique val already require this? Maybe something like DisallowUpdatesViaInsert?